src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,116 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final double DEFAULT_LOAD_FACTOR = 0.75f;
+    private static final int DEFAULT_INITIAL_CAPACITY = 1 << 4;
+    private static final int MAXIMUM_CAPACITY = 1 << 30;
+
+    private int size;
+    private int threshold;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (table == null) {
+            initialMyHash();
+        }
+        if (size >= threshold) {
+            resize();
+        }
+        int currentKeyHash = hash(key);
+        int i = getIndex(currentKeyHash);
+        Node<K, V> newNode = new Node<>(currentKeyHash, key, value, null);
+        if ((table[i]) == null) {
+            table[i] = newNode;
+            size++;
+            return;
+        }
+        putValueInTree(table[i], newNode);
     }
 
     @Override
     public V getValue(K key) {
+        if ((size == 0)) {
+            return null;
+        }
+        int currentKeyHash = hash(key);
+        int i = getIndex(currentKeyHash);
+        Node<K, V> currentNode;
+        if ((currentNode = table[i]) != null) {
+            do {
+                if ((currentNode.hash == currentKeyHash) && ((currentNode.key == key)
+                        || ((key != null) && key.equals(currentNode.key)))) {
+                    return currentNode.value;
+                }
+                currentNode = currentNode.next;
+            } while (currentNode != null);
+        }
         return null;
```

Comment:
```suggestion
        int index = getIndex(key);
        Node<K, V> node = table[index];
        while (node != null) {
            if (Objects.equals(node.key, key)) {
                return node.value;
            }
            node = node.next;
        }
        return null;
```

Reply:
do use "Objects.equals(obj1, obj2)" allowed when not possible to override equals() of obj1 ?